rotkehlchen/data/importer.py

```diff
@@ -182,3 +186,183 @@ def import_cointracking_csv(self, filepath: Path) -> None:
+            # those types are ignored because it doesn't affect the wallet balance
+            # or are not handled here
+            return
+        else:
+            raise UnsupportedCryptocomEntry(
+                f'Unknown entrype type "{row_type}" encountered during '
+                f'cryptocom data import. Ignoring entry',
+            )
+
+    # A swaping trade is a debited row + a credited row
+    def _import_cryptocom_swap(self, header: list, data: Any) -> None:
+        debited_row = None
+        credited_row = None
+        for data_row in data:
+            row = dict(zip(header, data_row))
+            if row['Transaction Kind'] == 'dynamic_coin_swap_debited':
+                debited_row = row
+            elif row['Transaction Kind'] == 'dynamic_coin_swap_credited':
+                credited_row = row
+
+        if credited_row is not None and debited_row is not None:
+            timestamp = deserialize_timestamp_from_date(
+                date=credited_row['Timestamp (UTC)'],
+                formatstr='%Y-%m-%d %H:%M:%S',
+                location='crypto.com',
+            )
+            notes = 'Coin Swap\nSource: crypto.com (CSV import)'
+            # No fees here since it's coin swaping
+            fee = Fee(ZERO)
+            fee_currency = A_USD
+
+            base_asset = Asset(credited_row['Currency'])
+            quote_asset = Asset(debited_row['Currency'])
+            pair = TradePair(f'{base_asset.identifier}_{quote_asset.identifier}')
+            base_amount_bought = deserialize_asset_amount(credited_row['Amount'])
+            quote_amount_sold = deserialize_asset_amount(debited_row['Amount'])
+            rate = Price(abs(base_amount_bought / quote_amount_sold))
+
+            trade = Trade(
+                timestamp=timestamp,
+                location=Location.CRYPTOCOM,
+                pair=pair,
+                trade_type=TradeType.BUY,
+                amount=base_amount_bought,
+                rate=rate,
+                fee=fee,
+                fee_currency=fee_currency,
+                link='',
+                notes=notes,
+            )
+            self.db.add_trades([trade])
+
+    def import_cryptocom_csv(self, filepath: Path) -> None:
+        with open(filepath, 'r', encoding='utf-8-sig') as csvfile:
+            data = csv.reader(csvfile, delimiter=',', quotechar='"')
+            self._import_cryptocom_swap(next(data), data)
+            # reset the iterator
+            csvfile.seek(0)
+            header = next(data)
+            for row in data:
+                try:
+                    self._consume_cryptocom_entry(dict(zip(header, row)))
+                except UnknownAsset as e:
+                    self.db.msg_aggregator.add_warning(
+                        f'During cryptocom CSV import found action with unknown '
+                        f'asset {e.asset_name}. Ignoring entry',
+                    )
+                    continue
+                except IndexError:
```

Comment:
As I said above, who raises the IndexError? What could definitely happen is a `KeyError`